extensions/amazon-lambda-rest/deployment/src/main/java/io/quarkus/amazon/lambda/http/deployment/AmazonLambdaHttpProcessor.java

```diff
@@ -80,4 +92,14 @@ public void generateScripts(OutputTargetBuildItem target,
         LambdaUtil.writeFile(target, "sam.native.yaml", output);
     }
 
+    @BuildStep()
+    @Record(STATIC_INIT)
+    public void setSecurityProvider(LambdaHttpBuildTimeConfig config,
+            RecorderContext context,
+            LambdaHttpRecorder recorder) {
+        if (config.identityProvider.isPresent()) {
+            recorder.setLambdaSecurityIdentityProvider(context.newInstance(config.identityProvider.get()));
```

Comment:
If you change this to use the existing security layer this should not be nessesary, users can already provide a custom IdentityProvider by just including a CDI bean.

Reply:
I don't think you still need this.